### PR TITLE
[FSSDK-9111] chore: prepare for 4.0.0-beta release

### PR DESCRIPTION
**Pull request by pulak-opti**
## Summary
- Prepare for 4.0.0-beta release

## Ticket
- [FSSDK-9111](https://jira.sso.episerver.net/browse/FSSDK-9111)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [4.0.0-beta] - May 10, 2023
+
+- Fix issue with parameters for send-odp-event. ([#386](https://github.com/optimizely/agent/pull/386))
+- Update go-sdk version to fix Null and Empty Values in sendOdpEvent. ([#385](https://github.com/optimizely/agent/pull/385))
+- Add tests for redis cache. ([#384](https://github.com/optimizely/agent/pull/384))
+- Add acceptance tests for send-odp-event. ([#383](https://github.com/optimizely/agent/pull/383))
+
```
Comment (yasirfolio3) [inline CHANGELOG.md:13]: This should be updated to reflect the new odp functionality that we're releasing similar to all of our other sdk's. [https://github.com/optimizely/go-sdk/pull/373/files](url). Also we might have to fix acceptance tests before releasing this

**Review by yasirfolio3 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,32 @@ All notable changes to this project will be documented in this file.
...
+- `FetchQualifiedSegments()` API has been added to the `/decide` endpoint. this API will retrieve user segments from the ODP server. The fetched segments will be used for audience evaluation. The fetched data will be stored in the local cache to avoid repeated network delays.
+
+- `SendOdpEvent()` API has been added with the `/send-opd-event` endpoint. Customers can build/send arbitrary ODP events that will bind user identifiers and data to user profiles in ODP.
+
+For details, refer to our documentation pages:
+
+* [Advanced Audience Targeting segment qualification methods](https://docs.developers.optimizely.com/feature-experimentation/v1.0/docs/advanced-audience-targeting-segment-qualification-methods-go) 
+
+* [Send Optimizely Data Platform data using Advanced Audience Targeting](https://docs.developers.optimizely.com/feature-experimentation/v1.0/docs/send-odp-data-using-advanced-audience-targeting-go)
```
Comment (Mat001) [inline CHANGELOG.md:23]: I think I would remove this reference from go-sdk. 

Instead I would add ref about server side sdk-s plus pages specific for Agent:
[Server SDK Support](https://docs.developers.optimizely.com/feature-experimentation/v1.0/docs/advanced-audience-targeting-for-server-side-sdks)

[Use Optimizely Agent](https://docs.developers.optimizely.com/feature-experimentation/docs/use-optimizely-agent)
[Configure Optimizely Agent](https://docs.developers.optimizely.com/feature-experimentation/docs/configure-optimizely-agent)
@yasirfolio3 What do you think.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,32 @@ All notable changes to this project will be documented in this file.
...
+* [Send Optimizely Data Platform data using Advanced Audience Targeting](https://docs.developers.optimizely.com/feature-experimentation/v1.0/docs/send-odp-data-using-advanced-audience-targeting-go) 
+
+### Bug Fix
+
+- Issue with parameter, null and empty values in `send-odp-event` have been fixed. ([#385](https://github.com/optimizely/agent/pull/385), [#386](https://github.com/optimizely/agent/pull/386))
+
+### Test Improvement
+
+- Tests for redis cache & acceptance tests for send-odp-event have been added. ([#383](https://github.com/optimizely/agent/pull/383), [#384](https://github.com/optimizely/agent/pull/384))
+
```
Comment (Mat001) [inline CHANGELOG.md:32]: Would also add this mention of breaking change:
image
@yasirfolio3 what u think?

**Review by Mat001 — CHANGES_REQUESTED** [review]

**Review by Mat001 — APPROVED** [review]
looks ok to me now

**Review by yasirfolio3 — APPROVED** [review]
LGTM

**Comment by Mat001** [comment]
@pulak-opti Before we release, let's fix the sourceclear issue

**Pull request merged**